lib/fh_cluster.js

```diff
@@ -62,12 +64,163 @@ module.exports = function fhCluster(workerFunc, optionalNumWorkers, optionalBack
     throw invalidWorkerTask.error;
   }
 
-  setOnWorkerListeningHandler();
-  setupOnExitBackOff(backoffStrategy, backoffResetTimeout);
-
-  start(workerFunc, numWorkers);
+  if (cluster.isMaster) {
+    startMaster( numWorkers, backoffStrategy, backoffResetTimeout );
+  } else {
+    startWorker( workerFunc, gracefullShutdownFn );
+  }
 };
 
+/********************************************/
```

Comment:
I like the master/node/common split...nice! :+1: 

Do you think there would be any value in splitting those different parts into separate files and requiring them here?